### PR TITLE
Added deprecation warning for gRPC MonitorService

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

**What kind of change does this PR introduce?**
Added deprecation warning for gRPC MonitorService

### CODE REVIEW

File: rpc/cc/arduino/cli/monitor/v1/monitor.proto [hunk 1]
```diff
@@ -21,7 +21,10 @@ option go_package = "github.com/arduino/arduino-cli/rpc/cc/arduino/cli/monitor/v
 
 import "google/protobuf/struct.proto";
 
-// MonitorService provides services for boards monitor
+// MonitorService provides services for boards monitor.
+// DEPRECATION WARNING: MonitorService is deprecated and will be removed in a
+// future release. Use CommandService.Monitor and
+// CommandService.EnumerateMonitorPortSettings instead.
```
Comment (per1234) [inline rpc/cc/arduino/cli/monitor/v1/monitor.proto:27]:
```suggestion
// future release. Use ArduinoCoreService.Monitor and
// ArduinoCoreService.EnumerateMonitorPortSettings instead.
```
Isn't this the service name?
https://arduino.github.io/arduino-cli/dev/rpc/commands/#cc.arduino.cli.commands.v1.ArduinoCoreService

File: rpc/cc/arduino/cli/monitor/v1/monitor.proto [hunk 2]
```diff
@@ -34,6 +37,9 @@ service MonitorService {
 // must contain a `monitor_config` message to initialize the monitor target.
 // All subsequent messages must contain bytes to be sent to the target
 // and must not contain a `monitor_config` message.
+// DEPRECATION WARNING: StreaminOpenRequest is deprecated and will be removed in
+// a future release. Use CommandService.Monitor and
+// CommandService.EnumerateMonitorPortSettings instead.
```
Comment (per1234) [inline rpc/cc/arduino/cli/monitor/v1/monitor.proto:42]:
```suggestion
// a future release. Use ArduinoCoreService.Monitor and
// ArduinoCoreService.EnumerateMonitorPortSettings instead.
```

File: rpc/cc/arduino/cli/monitor/v1/monitor.proto [hunk 3]
```diff
@@ -53,6 +59,9 @@ message StreamingOpenRequest {
 
 // Tells the monitor which target to open and provides additional parameters
 // that might be needed to configure the target or the monitor itself.
+// DEPRECATION WARNING: MonitorConfig is deprecated and will be removed
+// in a future release. Use CommandService.Monitor and
+// CommandService.EnumerateMonitorPortSettings instead.
```
Comment (per1234) [inline rpc/cc/arduino/cli/monitor/v1/monitor.proto:64]:
```suggestion
// in a future release. Use ArduinoCoreService.Monitor and
// ArduinoCoreService.EnumerateMonitorPortSettings instead.
```

File: rpc/cc/arduino/cli/monitor/v1/monitor.proto [hunk 4]
```diff
@@ -73,6 +82,9 @@ message MonitorConfig {
   int32 recv_rate_limit_buffer = 4;
 }
 
+// DEPRECATION WARNING: StreamingOpenResponse is deprecated and will be removed
+// in a future release. Use CommandService.Monitor and
+// CommandService.EnumerateMonitorPortSettings instead.
```
Comment (per1234) [inline rpc/cc/arduino/cli/monitor/v1/monitor.proto:87]:
```suggestion
// in a future release. Use ArduinoCoreService.Monitor and
// ArduinoCoreService.EnumerateMonitorPortSettings instead.
```

**Review by per1234 — CHANGES_REQUESTED** [review]

File: rpc/cc/arduino/cli/monitor/v1/monitor.proto (hunk 1 above)
Comment (cmaglie) [reply rpc/cc/arduino/cli/monitor/v1/monitor.proto:27]: 🤦🏼 yes correct

**Review by silvanocerza — APPROVED** [review]

**Review by per1234 — APPROVED** [review]

**Pull request merged**